R/r.R

```diff
@@ -66,6 +66,82 @@ use_test <- function(name = NULL, open = NULL) {
 }
 
 
+#' Automatically rename paired `R/` and `test/` files
+#'
+#' @description
+#' * Moves `R/{old}.R` to `R/{new}.R`
+#' * Moves `tests/testthat/test-{old}.R` to `tests/testthat/test-{new}.R`
+#' * Moves `tests/testthat/test-{old}-*.*` to `tests/testthat/test-{new}-*.*`
+#'   and updates paths in the test file.
+#' * Also removes now unneeded `context()` calls from the test file.
+#'
+#' This is a potentially dangerous operation, so you must be using Git in
+#' order to use this function
+#'
+#' @param old,new Old and new file names
+#' @export
+rename_files <- function(old, new) {
+  check_uses_git()
+
+  old <- path_ext_remove(old)
+  new <- path_ext_remove(new)
+
+  # Move .R file
+  r_old_path <- proj_path("R", old, ext = "R")
+  r_new_path <- proj_path("R", new, ext = "R")
+  if (file_exists(r_old_path)) {
+    ui_done("Moving {ui_path(r_old_path)} to {ui_path(r_new_path)}")
+    file_move(r_old_path, r_new_path)
+  }
+
+  if (!uses_testthat()) {
+    return()
+  }
+
+  # Move test files
+  rename_test <- function(path) {
+    file <- gsub(glue("^test-{old}"), glue("test-{new}"), path_file(path))
+    path(path_dir(path), file)
+  }
+  old_test <- dir_ls(proj_path("tests", "testthat"), glob = glue("*/test-{old}*"))
+  new_test <- rename_test(old_test)
+  if (length(old_test) > 0) {
+    ui_done("Moving {ui_path(old_test)} to {ui_path(new_test)}")
+    file_move(old_test, new_test)
+  }
+
+  # Update test file
+  test_path <- proj_path("tests", "testthat", glue("test-{new}"), ext = "R")
+  if (!file_exists(test_path)) {
+    return(invisible())
+  }
```

Comment:
I find it a little hard to follow the thread here. Could the test for "nothing to do / no files" and the early exit be earlier? I just can't immediately see the different cases of file existence / non-existence.

Reply:
This exits if there's no test file with the given name, as opposed to no tests at all.

Reply:
Ironically I now don't understand what this block of code does at all.